app/services/katello/pulp3/repository/ansible_collection.rb

```diff
@@ -21,15 +25,23 @@ def remote_class
         end
 
         def remotes_api
-          PulpAnsibleClient::RemotesCollectionApi.new(api_client)
+          self.class.remotes_api(smart_proxy)
+        end
```

Comment:
same thing here, this method can be centralized in the main pulp3/repository.rb class

Reply:
Also implementing that change to the ansible and docker repo type classes as well.